devicetypes/heltun/heltun-ht01-thermostat.src/heltun-ht01-thermostat.groovy

```diff
@@ -0,0 +1,538 @@
+		)
+		parameterMap().each { 
+			if (it.title != null) {
+				input (
+					title: "${it.title}",
+					description: it.description,
+					type: "paragraph",
+					element: "paragraph"
+				)
+			}
+			def unit = it.unit ? it.unit : ""
+			def defV = it.default as Integer
+			def defVDescr = it.options ? it.options.get(defV) : "${defV}${unit} - Default Value" 
+			input (
+				name: it.name,
+				title: null,
+				description: "$defVDescr",
+				type: it.type,
+				options: it.options,
+				range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+				defaultValue: it.default,
+				required: false
+			)
+		}
+	}
+}
+
+private channelNumber(String N) {
+	N.split(":")[-1] as Integer
+}
+
+def installed() {
+	state.oldLabel = device.label
+	def childName = "${device.displayName} Floor Temperature"
+	def existingChildren = getChildDevices()
+	def FloorTemperatureid = "${device.deviceNetworkId}:${1}"
+	def childExists = (existingChildren.find {child -> child.getDeviceNetworkId() == FloorTemperatureid} != NULL)
+	if (!childExists) {
+		addChildDevice("HE-TEMPERATURE", FloorTemperatureid, device.hubId,[completedSetup: true, label: childName, isComponent: true, componentName: "FloorTemperature", componentLabel: "FloorTemperature"])
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		return zwaveEvent(cmd)
+	}
+}
+
+def checkParam() {
+	boolean needConfig = false
+	parameterMap().each {
+		if (state."$it.name" == null || state."$it.name".state == "defNotConfigured") {
+			state."$it.name" = [value: it.default as Integer, state: "defNotConfigured"]
+			needConfig = true
+		}
+		if (settings."$it.name" != null && (state."$it.name".value != settings."$it.name" as Integer || state."$it.name".state == "notConfigured")) {
+			state."$it.name".value = settings."$it.name" as Integer
+			state."$it.name".state = "notConfigured"
+			needConfig = true
+		}
+	}
+	if ( needConfig ) {
+		configParam()
+	}
+}
+
+private configParam() {
+	def cmds = []
+	for (parameter in parameterMap()) {
+		if ( state."$parameter.name"?.value != null && state."$parameter.name"?.state in ["notConfigured", "defNotConfigured"] ) { 
+				cmds << zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$parameter.name".value, parameterNumber: parameter.paramNum, size: parameter.size).format()
+				cmds << zwave.configurationV2.configurationGet(parameterNumber: parameter.paramNum).format()
+				break
+		}
+	}
+	if (cmds) {
+		runIn(5, "checkParam")
+		sendHubCommand(cmds,500)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def parameter = parameterMap().find( {it.paramNum == cmd.parameterNumber } ).name
+	if (state."$parameter".value == cmd.scaledConfigurationValue){
+		state."$parameter".state = "configured"
+	}
+	else {
+		state."$parameter".state = "error"
+	}
+	configParam()
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
```

Comment:
is a versionGet ever sent?

Reply:
In fack, it is not sent.
I will remove it